pkg/kwok/controllers/stage_controller.go

```diff
@@ -345,18 +351,34 @@ func (c *StageController) playStage(ctx context.Context, resource *unstructured.
 }
 
 // patchResource patches the resource
-func (c *StageController) patchResource(ctx context.Context, resource *unstructured.Unstructured, patch []byte) (*unstructured.Unstructured, error) {
+func (c *StageController) patchResource(ctx context.Context, resource *unstructured.Unstructured, patch []byte, stage LifecycleStage) (*unstructured.Unstructured, error) {
 	logger := log.FromContext(ctx)
 	logger = logger.With(
 		"resource", log.KObj(resource),
 	)
 
```

Comment:
``` golang
	dynamicClient := c.dynamicClient
	if impersonator, ok := dynamicClient.(client.DynamicClientImpersonator); ok {
		...
	}
```

Reply:
modified the implementation based on this feedback with minor changes, PTL